Game.ino

```diff
@@ -80,17 +88,34 @@ void setup() {
 
 
 // checks if a movement button is the only one pressed
-bool onepressed(int button){
-  bool buttons[4] = {
-    arduboy.pressed(DOWN_BUTTON), arduboy.pressed(UP_BUTTON),
-    arduboy.pressed(RIGHT_BUTTON), arduboy.pressed(LEFT_BUTTON)
-  };
+void changedirection(){
+  // true when corresponding d-pad button is pressed
+  bool down = arduboy.pressed(DOWN_BUTTON);
+  bool up = arduboy.pressed(UP_BUTTON);
+  bool right = arduboy.pressed(RIGHT_BUTTON);
+  bool left = arduboy.pressed(LEFT_BUTTON);
+  
+  // false if more than one button is pressed
+  down = down && (!right && !left);
+  up = up && (!right && !left);
+  right = right && (!down && !up);
+  left = left && (!down && !up);
```

Comment:
Nice! :+1: 

Hmmm... interesting. This looks similar to my attempt here in the office. I suspected an issue with that `left` and `right` might not be "invalidated" because on multi-press `down` or `up` would be invalidated first. I would expect the issue to manifest itself as left/right turning also on multi-press, but I'm unable to trigger in the emulator.

Bonus points if you can tell my _why_ this works! :smile:

Reply:
I tried it myself, holding `up` and `right` at the same time results in the snake always moving right since vertical buttons are checked first, where `(!right && !left)` makes them always be false. Removing the checks for the right and left buttons makes no difference in the snake's movement.

Reply:
OK, it does what I suspected it does. :slightly_smiling_face:  I think I'm unable to replicate it because the ProjectABE emulator seems to be stuck with an old build.

This could be changed... but does this affect gameplay? What do you think?

Reply:
OK, I fixed the emulator issue for myself. I got to test the code now. I suppose we'd like to fix this as it can be called a "regression", where feature that worked now doesn't.

Do you have ideas?

Reply:
I fixed the issue by making new bools instead of overwriting the button press checks, so now `bool up` becomes `bool upbutton` for example, and `up = up && (!right && !left);` becomes `bool up = upbutton && (!rightbutton && !leftbutton);`.

Reply:
yup, that's the most obvious solution! :+1: But the code is getting lengthy, isn't it?

I can think of several more alternatives. And one of them I derived directly from your comment:
```c++
// false if more than one button is pressed
```

You could check how many buttons are being pressed. If mutiple are detected, exit early with `return`. Further, the check itself can be done in several ways -- some with their own caveats. :slightly_smiling_face:

Reply:
`if (down + up + right + left != 1) { return; }` seems to work fine, since in this case, `true` is counted as 1 and `false` is counted as 0. The direction no longer prioritizes left or right.

Reply:
Nice! :+1: